### ISSUE TITLE
Quick Order: Accessibility Pattern revision

### ISSUE THREAD

**Issue by developpeurweb** [SAP/spartacus#13654]
**Describe the bug**
- Although there is no keyboard trap, the Accessibility Pattern for this UI needs revision. Currently, the search box is triggered on "Enter" instead of using a button. Extra descriptions for screen reader users will be needed to convey functionality. Just to make sure this is what we want.
- There needs to be consistency among all internal search boxes so implementation is similar in all cases. E.g. Find a Store uses a button to trigger the search.
- Currently, all search boxes have different interaction patterns (product search, find a store, quick order), e.g. trigger mechanism, clear button, autocomplete
- This was already brought to the attention of @Manurod2018 and @vekterdude 

![Screen Shot 2021-09-08 at 4 25 48 PM](https://user-images.githubusercontent.com/15113219/132580957-0dc10e56-78c2-406c-90f6-e5d097803b66.png)

![2021-09-08_16-27-43](https://user-images.githubusercontent.com/15113219/132580974-4a803348-9b52-4f4d-b5a9-0cee422e5f80.jpg)

**Comment by bgambocjaviniar** [comment]
related https://github.com/SAP/spartacus/issues/13596

**Comment by dydome** [comment]
@developpeurweb may you please review this task as in 4.2 quick order has big upgrade

**Comment by developpeurweb** [comment]
@dydome 
Most of the issues still persist. I see the search field follows along the lines of the main search, therefore the "X" should be keyboard accessible and currently is not. Also, it does not convey functionality for screen readers as the main search does. 
This is more of a screen reader ticket, if too many breaking changes, then you may want to push it to 5.0

**Comment by KateChuen** [comment]
Tested on develop branch of spartacus using installation script
More extensive testing (on other screenreaders and devices) will be done with RC.

Have a few questions, can you confirm @developpeurweb 

- What I hear when I focus on the image is not meaningful to me. 
![QuickOrderImage](https://user-images.githubusercontent.com/45409515/149958221-27cbe935-53c9-446a-bacc-f6e61c4baa66.png)

- When I'm in read mode, and I have a text in the search field, although the result list is not displayed, it reads "5 products are available"
![quickOrderSearchList](https://user-images.githubusercontent.com/45409515/149958545-a53be685-0f11-427e-9542-b48e6142ed83.png)

I must say that it's consistent with the Main search though. But was just wondering: 
- In the search result list, when we tab on the first item, it says "Add Product .... List of 5 items". Shouldn't it say "List of 5 items" first, then "Add Product ..."? 
![QuickOrderList](https://user-images.githubusercontent.com/45409515/149958322-6dc8e1fd-bf1a-46ea-aa4b-ba7605539dd1.png)

- Also when we tab on a product from the list, it only says the product name. Shouldn't it also say the product ID and the price?
![QuickOrderSearchList1](https://user-images.githubusercontent.com/45409515/149958348-6efeb0ee-44eb-4fb5-9a30-ab2b7c49a8a7.png)

- In the list of quick order items, it would be nice to say how many items is there, because we can only add 10 items. So it would help if I could hear something like "List of x items", instead of having to keep count while it's reading the products to me.

**Comment by developpeurweb** [comment]
@KateChuen my answers below:

> What I hear when I focus on the image is not meaningful to me.

:red_circle: **This is a bug**. The `alt` tag is missing. Should be the same as on Cart Page.

> When I'm in read mode, and I have a text in the search field, although the result list is not displayed, it reads "5 products are available"

🟢  **This is OK**. If there is a string _456_ getting read, it informs there are 5 products, if user tabs backwards for interaction the products become available to the user.

> In the search result list, when we tab on the first item, it says "Add Product .... List of 5 items". Shouldn't it say "List of 5 items" first, then "Add Product ..."?

🟢  **This is OK**. It's VoiceOVer particular way to narrate elements. For NVDA narration says "list list with 5 items Add product Alpha 350 button".

> Also when we tab on a product from the list, it only says the product name. Shouldn't it also say the product ID and the price?

⚠️  Although not a bug, this **must be consistent with the main search.** 
@dydome this should be a **quick fix** I think: can you please remove the aria-label from the button (product) so divs and spans inside the button will get read as they are. If not a quick fix, then leave it as is.

> In the list of quick order items, it would be nice to say how many items is there, because we can only add 10 items. So it would help if I could hear something like "List of x items", instead of having to keep count while it's reading the products to me.

This is a great idea. But, we will need a service that updates the number of items in the Quick Order cart. Unfortunately, we're struggling with aria-live across the app. 

:red_circle: However, **things are missing and need to be fixed:** the limit hint is missing and the error message is not audible. @dydome can you please attenuate this by providing the following:
1. Complement the search box aria-label with the hint that: _"You can add up to 10 valid SKU per order"._
2. When the limit is reached, the message: "The product limit has been reached" should be wrapped with a role="alert".

![Screen Shot 2022-01-19 at 6 58 08 PM](https://user-images.githubusercontent.com/15113219/150240451-1de9ada5-1842-4554-ab4e-5d20c00dd11a.png)

**Comment by dydome** [comment]
@KateChuen @developpeurweb I made a PR with fixes for you -> https://github.com/SAP/spartacus/pull/14817 may you please review it and approve if it is fine?
